src/Resource/GetAllRate.php

```diff
@@ -38,17 +38,17 @@ public function fetch(ImportConnector $connector): \Iterator
         );
 
         $rates = $response['rates'];
-        $base = $response['asset_id_base'];
-        $quote = [];
-        $rate = [];
-        $time = [];
-
-        foreach($rates as $key => $value) {
-            $quote[] = $rates[$key]['asset_id_quote'];
-            $rate[] = $rates[$key]['rate'];
-            $time[] = $rates[$key]['time'];
-        }
-
-        return new AllRatesRecord($time, $base, $quote, $rate);
+
+        $rate = function () use ($rates) {
```

Comment:
This closure seems to just copy the array into an iterator. You can replace this entire code block with `new ArrayIterator($response['rates'])`.